R/available-data.R

```diff
@@ -104,128 +132,284 @@ spod_available_data_v1 <- function(
+    "0000-referencia\\/([0-9]{4})([0-9]{2})([0-9]{2})_",
+    "year=\\1\\/month=\\2\\/day=\\3\\/",
+    files_table$local_path
+  )
 
-# replace 2 digit day with 1 digit day
-files_table$local_path <- gsub("day=0([1-9])", "day=\\1", files_table$local_path)
+  # replace 2 digit month with 1 digit month
+  files_table$local_path <- gsub(
+    "month=0([1-9])",
+    "month=\\1",
+    files_table$local_path
+  )
 
-# change txt.gz to csv.gz
-files_table$local_path <- gsub("\\.txt\\.gz", "\\.csv\\.gz", files_table$local_path)
+  # replace 2 digit day with 1 digit day
+  files_table$local_path <- gsub(
+    "day=0([1-9])",
+    "day=\\1",
+    files_table$local_path
+  )
 
-# replace all municipal data download links with districts links
-# this is to address the bugs described in detail in:
-# http://www.ekotov.pro/mitma-data-issues/issues/011-v1-tpp-mismatch-zone-ids-in-table-and-spatial-data.html
-# http://www.ekotov.pro/mitma-data-issues/issues/012-v1-tpp-district-files-in-municipality-folders.html
-# the decision was to use distrcit data and aggregate it to replicate municipal data
-files_table$target_url <- gsub("mitma-municipios", "mitma-distritos", files_table$target_url)
-files_table$target_url <- gsub("mitma_municipio", "mitma_distrito", files_table$target_url)
+  # change txt.gz to csv.gz
+  files_table$local_path <- gsub(
+    "\\.txt\\.gz",
+    "\\.csv\\.gz",
+    files_table$local_path
+  )
 
-# add known file sizes from cached data
-file_sizes <- readr::read_csv(system.file("extdata", "url_file_sizes_v1.txt.gz", package = "spanishoddata"), show_col_types = FALSE)
-files_table <- dplyr::left_join(files_table, file_sizes, by = "target_url")
+  # replace all municipal data download links with districts links
+  # this is to address the bugs described in detail in:
+  # http://www.ekotov.pro/mitma-data-issues/issues/011-v1-tpp-mismatch-zone-ids-in-table-and-spatial-data.html
+  # http://www.ekotov.pro/mitma-data-issues/issues/012-v1-tpp-district-files-in-municipality-folders.html
+  # the decision was to use distrcit data and aggregate it to replicate municipal data
+  files_table$target_url <- gsub(
+    "mitma-municipios",
+    "mitma-distritos",
+    files_table$target_url
+  )
+  files_table$target_url <- gsub(
+    "mitma_municipio",
+    "mitma_distrito",
+    files_table$target_url
+  )
 
-# if there are files with missing sizes, impute them
-if (any(is.na(files_table$remote_file_size_mb))) {
-  # impute uknown file sizes
-  # primitive file categorisation
-  # Extract file category from the target URL
   files_table <- files_table |>
     dplyr::mutate(
-      file_category = stringr::str_extract(.data$target_url, "\\/maestra(\\d)-mitma-(distritos|municipios)\\/(ficheros-diarios|meses-completos)\\/")
+      study = dplyr::case_when(
+        grepl("maestra", .data$target_url) ~ "basic",
+        TRUE ~ ""
+      ),
+
+      type = dplyr::case_when(
+        grepl("maestra2", .data$target_url) ~ "number_of_trips",
```

Comment:
You could also use something like

```suggestion
        stringr::str_detect(target_url, "maestra2") ~ "number_of_trips",
```

For a 'tidier' implementation, no extra deps and {dplyr} depends on {stringr}. But the benefits would be very marginal if any so probably not worth implementing.